### PR TITLE
Cleanup master/slave terminology from documentation + Cleanup "Jenkins master" term from the warning in the Web UI

### PR DESCRIPTION
**Pull request by akatsadimas**
Partially addresses  #1465
Also updated Halm charts [link ](https://github.com/jenkinsci/configuration-as-code-plugin/compare/master...akatsadimas:cleanupMasterSlaveTerminology?expand=1#diff-ea15cec6747d8ff46bf708cbf047e931L6)since it has changed.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: plugin/pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <packaging>hpi</packaging>
 
   <name>Configuration as Code Plugin</name>
-  <description>Manage Jenkins master configuration as code</description>
+  <description>Manage Jenkins controller configuration as code</description>
```
Comment (oleg-nenashev) [inline plugin/pom.xml:17]: Maybe it's better to leave just "Jenkins" here. Agent configurations can be also managed on the controller side

**Review by oleg-nenashev — APPROVED** [review]
All changes look good, just one optional suggestion. Thanks for the PR!

File: plugin/pom.xml (hunk 1 above)
Comment (akatsadimas) [reply plugin/pom.xml]: addressed in 46affecd3f251ffca636ab1d75bfa8d10cc3849b

File: demos/kubernetes/README.md [hunk 2]
```diff
@@ -60,13 +60,13 @@ jenkins:
...
 
-          - name: "k8s-slave"
+          - name: "k8s-agent"
             namespace: "default"
             label: "linux-x86_64"
             nodeUsageMode: EXCLUSIVE
             containers:
               - name: "jnlp"
-                image: "jenkinsci/jnlp-slave:latest"
+                image: "jenkinsci/inbound-agent:latest"
```
Comment (timja) [inline demos/kubernetes/README.md:69]:
```suggestion
                image: "jenkins/inbound-agent:latest"
```

**Review by jetersen — COMMENTED** [review]
I am unsure whether controller is the correct term in some of the documentation. Perhaps `Jenkins server` is better understood than `Jenkins controller`

**Review by oleg-nenashev — APPROVED** [review]
> I am unsure whether controller is the correct term in some of the documentation. Perhaps Jenkins server is better understood than Jenkins controller

@jetersen Jenkins controller term was accepted as a replacement for "Jenkins master" in July 2020. There might be some concerns about "controller" vs. "agent" in some se-cases, but IMHO this pull request is a right step as a bulk cleanup.

**Comment by oleg-nenashev** [comment]
Taking +1 from @timja in the chat, I will go ahead and merge it

**Comment by oleg-nenashev** [comment]
Thanks a lot for the patch @akatsadimas !

**Pull request merged**